### PR TITLE
Skill Generation Rework

### PR DESCRIPTION
**Pull request by Windchild292**
This PR moves Skill Generation in MegaMek to multiple generators, adds a true constant generator that just returns the default skill level, replaces the RandomSkillsDialog with a panel based on Juliez's in the player dialog, and moves the magic numbers to enums.

It also standardizes the SkillLevel setup from None to Heroic, with Ultra-Green and Heroic being new additions to the standard defaults.

During this I discovered multiple bugs to do with the legacy gunnery option, the force piloting to be gunnery + 1 option, and Anti-'Mech infantry skill assignment, all of which I've fixed by standardizing how skills are assigned.

Requires https://github.com/MegaMek/mekhq/pull/2803

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java [hunk 1]
```diff
@@ -1564,7 +1560,7 @@ private void createArmy() {
     }
 
     public void loadRandomSkills() {
-        clientgui.getRandomSkillDialog().showDialog(clientgui.getClient().getGame().getEntitiesVector());
+
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java:1563]: What's the point of having this method if it's empty?

File: megamek/src/megamek/client/ui/swing/lobby/LobbyActions.java [hunk 2]
```diff
@@ -428,19 +428,9 @@ void setRandomSkills(Collection<Entity> entities) {
...
-                if (e.getCrew() instanceof LAMPilot) {
-                    skills = c.getRandomSkillsGenerator().getRandomSkills(e, true);
-                    ((LAMPilot) e.getCrew()).setGunneryAero(skills[0]);
-                    ((LAMPilot) e.getCrew()).setPilotingAero(skills[1]);
-                }
-            }
-            e.getCrew().sortRandomSkills();
+        for (final Entity entity : entities) {
+            final Client client = lobby.getLocalClient(entity);
+            client.getSkillGenerator().setRandomSkills(entity, true);
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/lobby/LobbyActions.java:433]: I like this unification of copy/pasted code.

File: megamek/src/megamek/common/enums/SkillLevel.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    /**
+     * This returns the default skill values by level. This should never return the value for NONE,
+     * as NONE means one does not have the skill.
+     * @return the default skill array pairing
+     */
+    public int[] getDefaultSkillValues() {
+        switch (this) {
+            case NONE:
+                MegaMek.getLogger().error("Attempting to get illegal default skill values for NONE Skill Level. Returning { 10, 10 }");
+                return new int[]{ 10, 10 };
```
Comment (NickAragua) [inline megamek/src/megamek/common/enums/SkillLevel.java:106]: Will 10,10 cause errors within the UI since the actual "maximum" is 8?

**Review by NickAragua — COMMENTED** [review]
Couple of questions, looks fine otherwise.

File: megamek/src/megamek/common/enums/SkillLevel.java (hunk 3 above)
Comment (Windchild292) [reply megamek/src/megamek/common/enums/SkillLevel.java:106]: Good point, and one I'd rather not test out. Swapped to 8.

File: megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/lobby/ChatLounge.java:1563]: I forgot to delete it. Completely unused.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**